pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/RealtimeSegmentConverter.java

```diff
@@ -82,14 +83,14 @@ public RealtimeSegmentConverter(MutableSegmentImpl realtimeSegment, String outpu
       List<String> invertedIndexColumns, List<String> noDictionaryColumns, List<String> varLengthDictionaryColumns,
       boolean nullHandlingEnabled) {
     this(realtimeSegment, outputPath, schema, tableName, tableConfig, segmentName, sortedColumn,
-        invertedIndexColumns, new ArrayList<>(), noDictionaryColumns, varLengthDictionaryColumns, nullHandlingEnabled);
+        invertedIndexColumns, Collections.emptyList(), noDictionaryColumns, varLengthDictionaryColumns, nullHandlingEnabled);
   }
 
   // Used in RealtimeSegmentConverterTest
   public RealtimeSegmentConverter(MutableSegmentImpl realtimeSegment, String outputPath, Schema schema,
```

Comment:
Let's remove the extra constructors and just keep one (easier to use).
The only usage of them is `RealtimeSegmentConverterTest.testNoVirtualColumnsInSchema()`, where you can make `RealtimeSegmentConverter.getUpdatedSchema()` static

Reply:
done